src/game/savegame/savegame.c

```diff
@@ -614,3 +616,30 @@ bool Savegame_RestartAvailable(int32_t slot_num)
     SAVEGAME_INFO *savegame_info = &m_SavegameInfo[slot_num];
     return savegame_info->features.restart;
 }
+
+GAMEFLOW_OPTION Savegame_PlayAvailableStory(int32_t slot_num)
+{
+    SAVEGAME_INFO *savegame_info = &m_SavegameInfo[slot_num];
+
+    int32_t gf_option = GF_START_GAME | g_GameFlow.first_level_num;
+
+    bool loop_continue = true;
```

Comment:
I have a feeling that this entire function body can be replaced with
```
SAVEGAME_INFO *savegame_info = &m_SaveGameInfo[slot_num];
GameFlow_StorySoFar(g_GameFlow.first_level_num, GFL_NORMAL, savegame_info->level_num);
return GF_EXIT_TO_TITLE;
```

Reply:
No because of things like:
```
        case GFS_EXIT_TO_LEVEL:
            return GF_START_GAME | ((int32_t)seq->data & ((1 << 6) - **1));**
```

Using your code snippet, only the Caves FMV plays, then it returns to title.

Reply:
OK fair, I understand it now. However `loop_continue` is not needed.

Reply:
Yes I'll just use `while(1)`. I was trying to keep it close to `Shell_Main` but that does:
```
        case GF_EXIT_GAME:
            loop_continue = false;
            break;
```
The `loop_continue` there is just visual anyway.